adapter.go

```diff
@@ -39,12 +39,19 @@ type Adapter struct {
 }
 
 // NewAdapter is the constructor for Adapter.
-// arg should be a PostgreS URL string or of type *pg.Options
-// The adapter will create a DB named "casbin" if it doesn't exist
+// arg should be a PostgreS URL string or of type *pg.Options or of type *pg.DB
+// If arg is not of type *pg.DB the adapter will create a DB named "casbin" if it doesn't exist
 func NewAdapter(arg interface{}) (*Adapter, error) {
```

Comment:
Please put the code into a new constructor called `NewAdapterByDb`.

Reply:
I don't want to mess with `arg interface{}` too much, as it got even harder to read.

Reply:
Yeah, I agree with that